tests/on-device/gnumake.bats

```diff
@@ -0,0 +1,8 @@
+@test 'GNU Make is functional' {
+  mkdir -p .tmp$$; pushd .tmp$$
+	  echo -e 'x:\n\techo desired output > x' > Makefile
+	  nix-shell -p gnumake --run 'make x'
+	  [[ -e x ]]
+	  [[ "$(cat x)" == 'desired output' ]]
+  popd; rm -r .tmp$$
```

Comment:
```suggestion
  mkdir -p .tmp-gnu-make
  echo -e 'x:\n\techo desired output > x' > Makefile
  nix-shell -p gnumake --run 'make --file .tmp-gnu-make/Makefile x'
  [[ -e x ]]
	[[ "$(cat x)" == 'desired output' ]]
  rm -r .tmp-gnu-make
```

pushd is not necessary, also I think it would help potential debugging if the directory has a more descriptive name :)

Reply:
amended